src/aks-preview/azext_aks_preview/_validators.py

```diff
@@ -593,3 +597,22 @@ def validate_enable_custom_ca_trust(namespace):
         if hasattr(namespace, 'os_type') and namespace.os_type != "Linux":
             raise ArgumentUsageError(
                 '--enable_custom_ca_trust can only be set for Linux nodepools')
+
+
+def validate_azure_keyvault_kms_key_vault_network_access(namespace):
+    key_vault_network_access = namespace.azure_keyvault_kms_key_vault_network_access
+    if key_vault_network_access:
+        err_msg = '--azure-keyvault-kms-key-vault-network-access is invalid. ' \
+                  'The valid options are Public and Private'
+
+        if key_vault_network_access != CONST_KEYVAULT_NETWORK_ACCESS_PUBLIC and key_vault_network_access != CONST_KEYVAULT_NETWORK_ACCESS_PRIVATE:
+            raise InvalidArgumentValueError(err_msg)
```

Comment:
Since you have defined the enum arg type for `azure_keyvault_kms_key_vault_network_access`, cli framework will make sure the value is valid. I think it's ok to remove this validator.

Reply:
removed.